layer/executables/layer_runtime.py

```diff
@@ -97,6 +104,50 @@ def _create_dataset(self, name: str, func: Callable[..., Any]) -> None:
         )
         data_catalog.store_dataset(func(), uuid.UUID(build_response.id.value))
 
+    def _create_model(self, name: str, func: Callable[..., Any]) -> None:
+        model = func()
+        model_flavor = get_flavor_for_model(model)
+        if model_flavor is None:
+            raise LayerFunctionRuntimeError("unsupported model flavor")
+        with tempfile.TemporaryDirectory() as model_dir:
+            model_flavor.save_model_to_directory(model, Path(model_dir))
+            model_catalog = ModelCatalogClient.create(self._client_config, self._logger)  # type: ignore
+            project = current_project_full_name()
+            asset_path = AssetPath(
+                name,
+                AssetType.MODEL,
+                org_name=project.account_name,  # type: ignore
+                project_name=project.project_name,  # type: ignore
+            )
+            source_digest = _get_function_source_digest(self._package_info.metadata)  # type: ignore
+            display_fabric = Fabric.F_LOCAL.value  # the fabric to display in the UI
+            model_version = model_catalog.create_model_version(
+                asset_path=asset_path,
+                description="",
+                source_code_hash=source_digest,
+                fabric=display_fabric,
+            )
+            model_train_id = model_catalog.create_model_train_from_version_id(
+                model_version.model_version.id
+            )
+            model_catalog.start_model_train(model_train_id)
+            model_catalog.complete_model_train(
+                model_train_id, model_flavor.PROTO_FLAVOR
+            )
+            model_storage_config = model_catalog.get_model_train_storage_configuration(
+                model_train_id
+            )
+            _upload_model_artifacts(Path(model_dir), model_storage_config)
+            model_catalog.store_training_metadata(
+                asset_name=asset_path.asset_name,
+                description="",
```

Comment:
where should we get this description from?

Reply:
From the `dataset` or `model` decorators. Will be mapped in a separate PR. https://linear.app/layer/issue/LAY-3647/capture-and-store-function-descriptions-in-the-runtime